### ISSUE TITLE
Parse messy version numbers involving tildes

### ISSUE THREAD

**Issue by RyanGlScott** [fosskers/versions#62]
On certain Ubuntu configurations, the output of `clang --version` will be something like this:

```
Ubuntu clang version 12.0.0-3ubuntu1~20.04.5
Target: x86_64-pc-linux-gnu
Thread model: posix
InstalledDir: /usr/bin
```

I'm interested in being able to parse the `12.0.0-3ubuntu1~20.04.5` bit. Currently, all of the parsers that the `versions` library offers (ranging from `versioning` to `mess`) will choke on this version due to the fact that it contains a tilde:

```
λ> case mess "12.0.0-3ubuntu1~20.04.5" of Left e -> error $ errorBundlePretty e
*** Exception: Mess:1:16:
  |
1 | 12.0.0-3ubuntu1~20.04.5
  |                ^
unexpected '~'
expecting '+', '-', '.', ':', '_', digit, end of input, letter, or white space
```

I doubt that this is a valid SemVer, but is there a way to make `mess` accept this?

**Comment by fosskers** [comment]
Ah interesting... this is an example of a "mess" that tantalizingly close to a SemVer. Let's see if `~` is truly illegal...

**Comment by fosskers** [comment]
> A pre-release version MAY be denoted by appending a hyphen and a series of dot separated identifiers immediately following the patch version. Identifiers MUST comprise only ASCII alphanumerics and hyphens [0-9A-Za-z-].

and

>  Examples: 1.0.0-alpha, 1.0.0-alpha.1, 1.0.0-0.3.7, 1.0.0-x.7.z.92, 1.0.0-x-y-z.--.

So that is indeed an illegal semver. Let's make it a `Mess` and see what happens!

**Comment by fosskers** [comment]
@RyanGlScott Can you test the linked PR?

**Comment by fosskers** [comment]
I just released `6.0.0` which includes this fix.
